extensions/azurecore/src/azurecore.d.ts

```diff
@@ -11,12 +11,64 @@ import { azureResource } from './azureResource/azure-resource';
  * IMPORTANT: THIS IS NOT A HARD DEFINITION unlike vscode; therefore no enums or classes should be defined here
  * (const enums get evaluated when typescript -> javascript so those are fine)
  */
-
-
 export const enum extension {
 	name = 'Microsoft.azurecore'
 }
 
+/**
+ * Enumeration of the Azure datacenter regions. See https://docs.microsoft.com/dotnet/api/microsoft.azure.management.resourcemanager.fluent.core.region
+ */
+export const enum AzureRegion {
+	australiacentral = 'australiacentral',
+	australiacentral2 = 'australiacentral2',
```

Comment:
i would love not to have to hardcode this list here. is there anything that we can do to expose this either from an npm package (or even at the sqltoolsservice layer)?

Reply:
I didn't look too hard but I don't know of anything that would get the correct localized string.

I'm not really concerned with hardcoding these - they'll never change (imagine how many problems that'd cause...) and we can easily add more as we encounter them. Sure it'd be great if there was an easy way to get the list but I'd rather not have to add another dependency on a library or do (relatively) expensive calls to STS just to get an enum value.

Reply:
Mmm. I was thinking more from the perspective of new regions, but ok.

Reply:
Yeah, but I think that'll be rare enough that it won't be a huge task to keep this updated.

We pretty much always hardcode location options anyways since trying to figure out what regions support the features we need isn't an easy task (you can query it through ARM I believe but in the end I just don't see a whole lot of value setting that up atm)

Reply:
I hear ya. Even if we did make use of an npm package, we'd have to update that when new regions go online with our lock files. I'm ok with this for now then.